### PR TITLE
Add formatting and linting

### PR DESCRIPTION
**Pull request by keithchev**
## Description

This PR adds formatting and linting tools for markdown files and a CI workflow to run them. The motivation for this is that we're going to add more material to this repo, so it would be nice to have a consistent style and avoid variation in formatting.

It also formats the extant markdown files; the changes are minimal.

Drive-by: delete empty markdown files (these seemed confusing).

### CODE REVIEW

File: .markdownlint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "default": true,
+    "line-length": false,
+    "no-inline-html": false,
+    "blanks-around-headers": true
+}
```
Comment (mertcelebi) [inline .markdownlint.json]: newline

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+v21.7.1
```
Comment (mertcelebi) [inline .nvmrc]: newline

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "bracketSpacing": true,
+    "overrides": [
+        {
+            "files": "*.md",
+            "options": {
+                "parser": "markdown"
+            }
+        }
+    ]
+}
```
Comment (mertcelebi) [inline .prettierrc]: newline

File: package.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    "lint": "markdownlint '**/*.md' --ignore node_modules",
+    "lint-fix": "markdownlint '**/*.md' --fix --ignore node_modules",
+    "format-check": "prettier --check '**/*.md'",
+    "format": "prettier --write '**/*.md'"
+  },
+  "devDependencies": {
+    "markdownlint-cli": "^0.40.0",
+    "prettier": "^3.2.5"
+  }
+}
```
Comment (mertcelebi) [inline package.json]: newline

**Review by mertcelebi — COMMENTED** [review]
Two notes:
* I don't know if we need both of these tools. Given `prettier` does both `check` and `fix`, do we need a markdown specific linter?
* Don't forget to add newlines to the missing files

**Review by mertcelebi — APPROVED** [review]

**Comment by keithchev** [comment]
> * I don't know if we need both of these tools. Given `prettier` does both `check` and `fix`, do we need a markdown specific linter?

We discussed offline but just for the record here, it turns out that `prettier` doesn't check for or fix certain markdown-specific syntax issues like incorrect headers (e.g. `#Header` instead of `# Header`), so we need to use both `prettier` and `markdownlint`.

**Pull request merged**